routing/single_mwm_router.cpp

```
@@ -215,6 +197,46 @@ bool SingleMwmRouter::LoadIndex(MwmSet::MwmId const & mwmId, string const & coun
   }
 }
 
+void SingleMwmRouter::BuildRoute(MwmSet::MwmId const & mwmId, IndexGraphStarter & starter, vector<Joint::Id> const & joints, RouterDelegate const & delegate, Route & route) const
+{
+  vector<RoutePoint> routePoints;
+  starter.RedressRoute(joints, routePoints);
+
+  vector<Junction> junctions;
+  junctions.reserve(routePoints.size());
+
+  Geometry & geometry = starter.GetGraph().GetGeometry();
+  // TODO: Use real altitudes for pedestrian and bicycle routing.
+  for (RoutePoint const & routePoint : routePoints)
+    junctions.emplace_back(geometry.GetPoint(routePoint.GetRoadPoint()), feature::kDefaultAltitudeMeters);
+
+  shared_ptr<traffic::TrafficInfo::Coloring> trafficColoring = m_trafficCache.GetTrafficInfo(mwmId);
+  ReconstructRoute(m_directionsEngine.get(), m_roadGraph, trafficColoring, delegate, junctions, route);
+
+  // ReconstructRoute duplicates all points except start and finish.
+  // Therefore one needs fix time indexes to fit reconstructed polyline.
+  // TODO: rework ReconstructRoute and remove this stuff.
+  if (routePoints.size() >= 2 && route.GetPoly().GetSize() + 2 == routePoints.size() * 2)
```

Да, сломался.
Но хлама в маршруте нет. Это опциональный шаг корректировки времени, который можно целиком убрать и все равно будет работать.
Если что-то сломалось, это необязательно доламывать дальше.

И почему всего лишь false вернуть и маршрут поломать?
Это же ИНВАРИАНТ СЛОМАЛСЯ! Ужас то какой.
Давай тогда уж грохнем весь application с помощью CHECK.
Хотя это тоже мелко, надо удалить приложение с телефона, вдруг какие-то файлы от этого испортились?
Взорвать телефон?
Убить пользователя?
Накрыть ядерной бомбой место происшествия?

Если серьезно, то надо уметь разумно выбирать между fail fast и fail safe подходами в зависимости от контекста, а не всегда сразу хвататься за fail fast, просто потому что так больше нравится. Предлагаю попробовать отложить технические вкусы и посмотреть на это глазами пользователя: какой вариант развития событий лучше для него?